### ISSUE TITLE
TypeScript types for `serialize` property of `EnhancerOptions` seem wrong

### ISSUE THREAD

**Issue by blaiz** [reduxjs/redux-toolkit#751]
Hello,

According to https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#serialize the `serialize` property can be assigned an object with the following properties: `options`, `replacer`, `reviver`, `immutable` and `refs`.

However, at https://github.com/reduxjs/redux-toolkit/blob/a1e8becbb2e4653d721bdc1d2363a45562c526a2/src/devtoolsExtension.ts#L39 the type definition for `serialize` is defined as follows:
```ts
  serialize?:
    | boolean
    | {
        date?: boolean
        regex?: boolean
        undefined?: boolean
        error?: boolean
        symbol?: boolean
        map?: boolean
        set?: boolean
        function?: boolean | Function
      }
```
which looks to me like it's skipping one level and it's defining the types for the `options` properties directly inside `serialize`. Seems like it should look more like:
```ts
  serialize?:
    | boolean
    | {
        options: | boolean
      | {
          date?: boolean
          regex?: boolean
          undefined?: boolean
          error?: boolean
          symbol?: boolean
          map?: boolean
          set?: boolean
          function?: boolean | Function
        },
      replacer: () => {}, // with some proper types here
      reviver: () => {} // etc. with the other properties listed above
      }
```

Thoughts?
If you agree, I can try and do some research on what the type definition for these missing properties would be and whip up a PR perhaps.

Thanks

**Comment by markerikson** [comment]
Hmm.  Yeah, I copy-pasted some typedefs over, and it's entirely possible I screwed that up somewhere :)  If you could look into that, I'd appreciate it.  Thanks!

**Comment by blaiz** [comment]
Sounds good. I found some types at https://www.github.com/reduxjs/redux-devtools/tree/master/packages%2Fredux-devtools-serialize%2Fsrc%2Fimmutable%2Fserialize.ts

I'll take a closer look tomorrow.

**Comment by markerikson** [comment]
Fixed by #752 .
